### ISSUE TITLE
Labels in OpenSSL dataset

### ISSUE THREAD

**Issue by AthaSSiN** [gadiluna/SAFE#2]
Hi,

We couldn't find any field corresponding to labels in the OpenSSL dataset (as mentioned by you in your publication). Can you please explain how to extract/compute them? We were looking at the AMD64MultipleCompilers.db dataset.

We are planning to perform supervised learning along with similarity measures (similar to what you did in the paper).

My colleague has mailed Luca Massarelli regarding the same.

Thanks in advance!

**Comment by gadiluna** [comment]
The db contains a table x_pairs containing the pairs for each split, with x in (train,test, validation). 

Each entry of this table has structure
ID, TRUE_PAIR, FALSE_PAIR.

Each entry represents an entire epoch, TRUE_PAIR is a list of pairs of IDs, each pair in this list is a pair with implicit label +1 (similar functions). FALSE_PAIR is the same for the label -1 (dissimilar functions). 

The IDs can be used to lookup the function details using the function table.

**Comment by AthaSSiN** [comment]
We tried using these true pairs and false pairs and we even experimented a bit with the DataSplitter class to understand its working, however we couldn't figure out how to find if a function comes from a benign / malware program (or any other label like it being an encryption /sorting etc function).

When looking up function details, we saw that in a database, there are several functions with the same details (like compiler, program and function_name), and with different ids. Ideally we think these should come up in true pairs, but they don't seem to use any labels like benign / malware or function types (like encryption /sorting etc).

Kindly help us identify the issue in our understanding of the db. 

Thanks!
